Tweak/Helper/KayokoHelper.xm

```diff
@@ -132,27 +140,133 @@ static void override_UIKeyboardLayoutStar_didMoveToWindow(UIKeyboardLayoutStar*
+        return %orig(@"doc.on.clipboard.fill", arg2);
+    return %orig;
+}
+
+- (void)setImageName:(NSString *)arg1 {
+    if ([arg1 isEqualToString:@"mic"])
+        return %orig(@"doc.on.clipboard");
+    else if ([arg1 isEqualToString:@"mic.fill"])
+        return %orig(@"doc.on.clipboard.fill");
+    return %orig;
+}
+
+%end
+
+%hook UIKeyboardDockItemButton
+
+- (CGRect)imageRectForContentRect:(CGRect)arg1 {
+    CGRect origRect = %orig;
+    if (ABS(origRect.size.width - origRect.size.height) > 1.0) {
+        // adjust image to its 83% but keep its center
+        CGSize newSize = CGSizeMake(origRect.size.width * 0.833, origRect.size.height * 0.833);
+        CGPoint newOrigin = CGPointMake(origRect.origin.x + (origRect.size.width - newSize.width) / 2, origRect.origin.y + (origRect.size.height - newSize.height) / 2);
+        return CGRectMake(newOrigin.x, newOrigin.y, newSize.width, newSize.height);
+    }
+    return origRect;
+}
+
+%end
+
+%end
+
 #pragma mark - Notification callbacks
 
+static BOOL _kApplicationIsInForeground = YES;
+
+%group AppEvents
+
+%hook UIKeyboardImpl
++ (void)applicationDidBecomeActive:(id)arg1 {
+    _kApplicationIsInForeground = YES;
+    %orig;
+}
++ (void)applicationWillResignActive:(id)arg1 {
+    _kApplicationIsInForeground = NO;
+    %orig;
+}
+%end
+
+%end
+
+@interface UIKeyboardTaskQueue : NSObject
+- (void)addTask:(id)arg1;
+- (void)performSingleTask:(id)arg1;
+@end
+
+@interface UIKBInputDelegateManager (Private)  // iOS 15
+- (void)insertText:(id)arg1 updateInputSource:(BOOL)arg2;
+- (void)clearForwardingInputDelegateAndResign:(BOOL)arg1;
+@end
+
+@interface UIKeyboardImpl (Private)
+- (void)addInputString:(NSString *)arg1 withFlags:(unsigned long long)arg2 executionContext:(id)arg3;
+- (void)completeAddInputString:(NSString *)arg1;
+- (void)completeAddInputString:(NSString *)arg1 generateCandidates:(BOOL)arg2;
+- (void)clearForwardingInputDelegateAndResign:(BOOL)arg1;
+- (void)updateReturnKey;
+- (UIKBInputDelegateManager *)inputDelegateManager;
+- (UIKeyboardTaskQueue *)taskQueue;
+@end
+
 static void paste() {
+    if (!_kApplicationIsInForeground) return;
     UIPasteboard* pasteboard = [UIPasteboard generalPasteboard];
     if ([pasteboard hasStrings]) {
-        if (iOS15) {
-            UIKBInputDelegateManager* delegateManager = [[objc_getClass("UIKeyboardImpl") activeInstance] inputDelegateManager];
-            [delegateManager insertText:[pasteboard string]];
+        NSString *pbs = [pasteboard string];
+        UIKeyboardImpl *kb = [%c(UIKeyboardImpl) activeInstance];
+        if (@available(iOS 15.0, *)) {
+            UIKBInputDelegateManager* delegateManager = [kb inputDelegateManager];
+            [delegateManager insertText:pbs];
+            if ([delegateManager respondsToSelector:@selector(clearForwardingInputDelegateAndResign:)])
+                [delegateManager clearForwardingInputDelegateAndResign:YES];
+            [kb updateReturnKey];
```

Comment:
Bugfix: this is a workaround for input methods for asian languages.

Reply:
What's the issue that asian languages have?

Reply:
If you paste something directly from Kayoko, the asian input method will receive the string.